### ISSUE TITLE
slf4j-nop dependency

### ISSUE THREAD

**Issue by meredrica** [micronaut-projects/micronaut-openapi#908]
### Issue description

Hello!

The project depends on `org.slf4j:slf4j-nop:jar:1.7.36`, this can result in multiple logger bindings.
This can of course be worked around by excluding the dependency.  My question is why this is added in the first place?
It seemingly was introduced in #822.
Depending on `slf4j-api` should suffice I think, but I only glanced over the code quickly.

**Comment by altro3** [comment]
@meredrica If you have another solution to fix this: https://github.com/micronaut-projects/micronaut-openapi/issues/821 - PR is welcome! :-)

**Comment by graemerocher** [comment]
ideally the correct fix is to not use code that initializes logging in the compiler

**Comment by jjathman** [comment]
Have to agree that this fix was worse than the previous problem. The previous issue was a warning, although annoying, it did not break applications. This seems like a breaking issue for applications which use this module. Updating to Micronaut 3.8.x causes logging to completely break without doing the workaround of removing the SLF4j No-op implementation.

**Comment by graemerocher** [comment]
yeah the solution to the problem is not a viable one. Build time only modules shouldn't include code paths that initialize logging and these need to be removed to fix this issue for good

**Comment by altro3** [comment]
@graemerocher But problem in micronaut-inject. micronaut-openapi by default tries to use environments and tries to read the application.yml configuration files, which uses the micronaut-inject logic that slf4j uses. No other dependency on micronaut-openapi and micronaut-openapi itself uses and initializes slf4j . I suspect that the initialization happens automatically when the processor annotation is run.

**Comment by graemerocher** [comment]
yeah the module is probably using an API in `micronaut-inject` that is designed to be used at runtime not at compilation time

**Comment by jjathman** [comment]
One thing that is confusing to me is why these dependencies are even causing runtime problems when they are being included as `compileOnly`. My expectation is that the `nop` dependency would not be available.
